source/Worlds/ManualWorld.hpp

```diff
@@ -0,0 +1,378 @@
+/**
+ * This file is part of the Fall 2023, CSE 491 course project.
+ * @brief A World that consists of trees, water, and grass cells.
+ * @note Status: PROPOSAL
+ **/
+
+#pragma once
+
+#include <cassert>
+#include <stdlib.h>
+
+#include "../core/WorldBase.hpp"
+
+namespace cse491_team8 {
+
+  class ManualWorld : public cse491::WorldBase {
+  protected:
+    enum ActionType { REMAIN_STILL=0, MOVE_UP, MOVE_DOWN, MOVE_LEFT, MOVE_RIGHT, ATTACK };
+    enum FacingDirection { UP=1, RIGHT, DOWN, LEFT};
+
+    size_t grass_id;  ///< Easy access to floor CellType ID.
+    size_t tree_id;   ///< Easy access to tree CellType ID.
+    size_t water_id;  ///< Easy access to water CellType ID.
+    size_t bridge_id; ///< Easy access to bridge CellType ID.
+    size_t rock_id;   ///< Easy access to rock CellType ID.
+
+    /// Provide the agent with movement actions.
+    void ConfigAgent(cse491::AgentBase & agent) override {
+      agent.AddAction("up", MOVE_UP);
+      agent.AddAction("down", MOVE_DOWN);
+      agent.AddAction("left", MOVE_LEFT);
+      agent.AddAction("right", MOVE_RIGHT);
+      agent.AddAction("attack", ATTACK);
+    }
+
+  public:
+    ManualWorld() {
+      grass_id = AddCellType("grass", "Grass that you can easily walk over.", ' ');
+      tree_id = AddCellType("tree", "Tree that you cannot pass without an axe.", '^');
+      water_id = AddCellType("water", "Water that cannot be crossed without a boat.", '~');
+      bridge_id = AddCellType("bridge", "Bridge that allows the playerto cross water.", '#');
+      rock_id = AddCellType("rock", "Rock that the player cannot cross", '$');
+      main_grid.Read("../assets/grids/team8_grid_large.grid", type_options);
+    }
+    ~ManualWorld() = default;
+
+
+    /// @brief Checks the strength between two agents
+    /// @param other_agent The autonomous agent to compare
+    /// @param agent The interface (player) agent to compare
+    /// Prints the stronger agent and removes the weaker
```

Comment:
The world objects shouldn't be printing anything -- they should allow the interface to do this.  When we are using a graphical, network, or web interface, couts won't be visible.  This DOES point out that we need some mechanism for messages to be sent to a user.